packages/flutter/test/widgets/animated_switcher_test.dart

```diff
@@ -56,6 +56,141 @@ void main() {
     await tester.pumpAndSettle();
   });
 
+  testWidgets('animation.status in transitionBuilder dictates correct transition phase', (WidgetTester tester) async {
```

Comment:
Can you add more tests to verify that the transition builder is not called more often than necessary? E.g. it should only be called once for transitioning in and once for transitioning out? The current test doesn't show if it is called once for dismissed and once for forward...